src/golang/lib/workflow/operator/base.go

```diff
@@ -273,15 +296,18 @@ func (bfo *baseFunctionOperator) jobSpec(fn *function.Function) job.Spec {
 		outputArtifactTypes = append(outputArtifactTypes, outputArtifact.Type())
 	}
 
-	return &job.FunctionSpec{
+	bfo.cachedSpec = &job.FunctionSpec{
 		BasePythonSpec: job.NewBasePythonSpec(
 			job.FunctionJobType,
 			bfo.jobName,
 			*bfo.storageConfig,
 			bfo.metadataPath,
 		),
 		FunctionPath: fn.StoragePath,
-		/* `FunctionExtractPath` is set by the job manager at launch time. */
+
+		// TODO(kenxu): we always use the default storage directory here.
+		FunctionExtractPath: path.Join(job.DefaultOperatorStorageDir, uuid.New().String()),
```

Comment:
any ideas for how to grab the correct path here (we need to use the same path that the job manager is using)? The job manager is embedded very tightly with the AqServer so I don't see a great way of doing this at the moment. However, the rest of the code definitely seems like its at the right level of abstraction.